spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -1425,18 +1432,12 @@ def refresh_container(self, give_focus=False):
                 client.set_info_page()
                 client.shellwidget.hide()
                 client.layout.addWidget(self.infowidget)
-                self.infowidget.show()
+                if self.infowidget is not None:
```

Comment:
There is another access of `self.infowidget` that is not protected by this validation. It's in the `update_font` method:

https://github.com/spyder-ide/spyder/blob/7ea1f61e418f07af784b10fdf13ebeee0e5e4560/spyder/plugins/ipythonconsole/widgets/main_widget.py#L1394-L1395

Please change that to be

```python
if self.enable_infowidget and self.infowidget is not None:
    ...
```